src/v/kafka/server/connection_context.cc

```diff
@@ -72,6 +72,14 @@ ss::future<> connection_context::process_one_request() {
                       return ss::make_ready_future<>();
                   }
                   return dispatch_method_once(std::move(h.value()), s)
+                  .handle_exception_type(
+                    [this](const kafka_api_version_not_supported_exception& e) {
+                        vlog(
+                          klog.warn,
+                          "Error while processing request from {} - {}",
+                          _rs.conn->addr,
+                          e.what());
```

Comment:
It looks to me like this doesn't drop the connection now, and i'm pretty sure that is what we want to do. I'd probably choose to move this to `protocol::apply(net::server::resources rs)` or rethrow the exception here so that the connection is shutdown.

https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaApis.scala#L168